plugins/c8y_remote_access_plugin/src/auth.rs

```diff
@@ -0,0 +1,22 @@
+use c8y_api::http_proxy::C8yMqttJwtTokenRetriever;
+use miette::IntoDiagnostic;
+use tedge_config::TEdgeConfig;
+
+pub struct Jwt(String);
+
+impl Jwt {
+    pub fn authorization_header(&self) -> String {
+        format!("Bearer {}", self.0)
+    }
+
+    pub async fn retrieve(config: &TEdgeConfig) -> miette::Result<Jwt> {
+        let mut retriever =
+            C8yMqttJwtTokenRetriever::from_tedge_config(config).into_diagnostic()?;
+
+        retriever
+            .get_jwt_token()
+            .await
+            .map(|resp| Jwt(resp.token()))
+            .into_diagnostic()
+    }
+}
```

Comment:
Do we have any built-in retries if the initial retrieval of the JWT fails?

Reply:
Yep, this is using the same logic as the rest of the mapper for retries (although I suspect in general the 20s wait will be longer than useful here). That said, I expect the retries to generally be useful when starting things up for the first time, where we might not have a c8y connection, rather than in this case where we must have a connection as we were triggered from an operation.

Reply:
Ok, that sounds fine. And yes the 20s wait would cause a timeout on the server side (in the remote access microservice), but we can ignore this use-case...and once Cumulocity IoT HTTP requests support certificate authentication, then the JWT might become less relevant.